coverage/io/io.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io
```

Comment:
I don't think it avoids error handling, it's handling the errors in this common place.  Plus, I think the logic in Write function is complicated enough to be in its own function

Reply:
Again, calling `fatal` deep within application logic is incredibly unidiomatic for go. instead things that can error pass errors up, and if the caller wants to handle or swallow the error it can. if the error is propogated to `main()`, _then_ fatal is called.

Here's a more idiomatic example nobody here was involved in: https://github.com/GoogleCloudPlatform/k8s-cluster-bundle/blob/c64a7255ba5a07bf631dcfb1907dbeabe54c0a09/cmd/bundlectl/main.go#L31

Or this post, from a well known contributor to the go community: https://dave.cheney.net/2015/11/05/lets-talk-about-logging
> Don’t log at fatal level, prefer instead to return an error to the caller. If the error bubbles all the way up to main.main then that is the right place to handle any cleanup actions before exiting.